Mathlib/Data/Polynomial/UnitTrinomial.lean

```diff
@@ -181,156 +181,165 @@ theorem isUnitTrinomial_iff :
       trinomial_middle_coeff hkm hmn, trinomial_trailing_coeff' hkm hmn]
-    simp_rw [← Units.val_pow_eq_pow_val, Int.units_sq, Units.val_one, ← add_assoc, bit1, bit0]
+    simp_rw [← Units.val_pow_eq_pow_val, Int.units_sq]
   · have key : ∀ k ∈ p.support, p.coeff k ^ 2 = 1 := fun k hk =>
       Int.sq_eq_one_of_sq_le_three
-        ((single_le_sum (fun k hk => sq_nonneg (p.coeff k)) hk).trans hp.le) (mem_support_iff.mp hk)
-    refine' is_unit_trinomial_iff.mpr ⟨_, fun k hk => isUnit_ofPowEqOne (key k hk) two_ne_zero⟩
-    rw [sum_def, sum_congr rfl key, sum_const, Nat.smul_one_eq_coe] at hp 
+        ((single_le_sum (fun k _ => sq_nonneg (p.coeff k)) hk).trans hp.le) (mem_support_iff.mp hk)
+    refine' isUnitTrinomial_iff.mpr ⟨_, fun k hk => isUnit_ofPowEqOne (key k hk) two_ne_zero⟩
+    rw [sum_def, sum_congr rfl key, sum_const, Nat.smul_one_eq_coe] at hp
     exact Nat.cast_injective hp
 #align polynomial.is_unit_trinomial_iff' Polynomial.isUnitTrinomial_iff'
 
 theorem isUnitTrinomial_iff'' (h : p * p.mirror = q * q.mirror) :
     p.IsUnitTrinomial ↔ q.IsUnitTrinomial := by
-  rw [is_unit_trinomial_iff', is_unit_trinomial_iff', h]
+  rw [isUnitTrinomial_iff', isUnitTrinomial_iff', h]
 #align polynomial.is_unit_trinomial_iff'' Polynomial.isUnitTrinomial_iff''
 
 namespace IsUnitTrinomial
 
 theorem irreducible_aux1 {k m n : ℕ} (hkm : k < m) (hmn : m < n) (u v w : Units ℤ)
-    (hp : p = trinomial k m n u v w) :
-    C ↑v * (C ↑u * X ^ (m + n) + C ↑w * X ^ (n - m + k + n)) =
+    (hp : p = trinomial k m n (u : ℤ) v w) :
+    C (v : ℤ) * (C (u : ℤ) * X ^ (m + n) + C (w : ℤ) * X ^ (n - m + k + n)) =
       ⟨Finsupp.filter (Set.Ioo (k + n) (n + n)) (p * p.mirror).toFinsupp⟩ := by
   have key : n - m + k < n := by rwa [← lt_tsub_iff_right, tsub_lt_tsub_iff_left_of_le hmn.le]
   rw [hp, trinomial_mirror hkm hmn u.ne_zero w.ne_zero]
   simp_rw [trinomial_def, C_mul_X_pow_eq_monomial, add_mul, mul_add, monomial_mul_monomial,
-    to_finsupp_add, to_finsupp_monomial, Finsupp.filter_add]
+    toFinsupp_add, toFinsupp_monomial, Finsupp.filter_add]
+  -- Porting note: added next line (less powerful `simp`).
+  rw [Finsupp.filter_add, Finsupp.filter_add, Finsupp.filter_add, Finsupp.filter_add,
+    Finsupp.filter_add, Finsupp.filter_add, Finsupp.filter_add, Finsupp.filter_add]
   rw [Finsupp.filter_single_of_neg, Finsupp.filter_single_of_neg, Finsupp.filter_single_of_neg,
     Finsupp.filter_single_of_neg, Finsupp.filter_single_of_neg, Finsupp.filter_single_of_pos,
     Finsupp.filter_single_of_neg, Finsupp.filter_single_of_pos, Finsupp.filter_single_of_neg]
-  · simp only [add_zero, zero_add, of_finsupp_add, of_finsupp_single]
-    rw [C_mul_monomial, C_mul_monomial, mul_comm ↑v ↑w, add_comm (n - m + k) n]
-  · exact fun h => h.2.Ne rfl
+  · simp only [add_zero, zero_add, ofFinsupp_add, ofFinsupp_single]
+    -- Porting note: added next two lines (less powerful `simp`).
```

Comment:
I know it's late in the day, but it's super helpful if we include the original proof step in the tactic, so to diagnose regressions we don't have to open mathlib3 and find the corresponding point.

Reply:
The original code is right there, I just added new code, so if simp becomes stronger again, the rewrites will fail.